public/pages/AnomalyCharts/containers/AnomaliesChart.tsx

```diff
@@ -172,6 +177,17 @@ export const AnomaliesChart = React.memo((props: AnomaliesChartProps) => {
     );
   };
 
+  const hasValidHCProps = () => {
+    return (
+      props.isHCDetector &&
+      props.onHeatmapCellSelected &&
+      props.detectorCategoryField &&
+      (props.showAlerts !== true ||
+        (props.showAlerts &&
+          props.onDisplayOptionChanged &&
+          props.entityAnomalySummaries))
+    );
```

Comment:
Yeas, your understanding is right. As per above [comment](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/pull/350#discussion_r547009140), I will change from `props.showAlerts` to `props.isNotSample` after Tyler's change is merged. 

More context is: for now `props.showAlerts` is a flag mainly for indicating whether the `chart` is used for Preview result case, aka Sample result case. If false, it is for Sample: if yes, it is for Real detection result. It has such name because we only show alert button on real detection result page, not on Sample result one. But this is not appropriate name because such flag is being used for 2 purposes(determining if chart is for Sample, and if chart should show alert button). That's why Tyler has [change](https://github.com/ohltyler/anomaly-detection-kibana-plugin/commit/33b4c0c73364e6ed86dcfa4102032134d6d7fb49) to refactor use of `showAlerts`, and I will use new `props. isNotSample` once his change is checked in.

Reply:
got it.  Thanks for the explanation.